controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -47,6 +47,7 @@ type dogstatsdFeature struct {
 
 	useHostNetwork         bool
 	originDetectionEnabled bool
+	tagCardinality         string
```

Comment:
suggestion: use type `*TagCardinalityName` here, and convert to string when setting the envvar below